### PR TITLE
[W10H01] Add tests for `TrialOfTheGrasses`

### PR DESCRIPTION
**Pull request by nilsreichardt**
* Test of the example from `main()`
* Test with a single node

### CODE REVIEW

File: w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package pgdp.trials;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+public class TrailOfTheGrassesTest {
```
Comment (Anatoly03) [inline w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java:9]:
```suggestion
public class TrialOfTheGrassesTest {
```
typo

**Review by Anatoly03 — CHANGES_REQUESTED** [review]
Trail of Grasses sounds like a pathways human have made

File: w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Test
+    void testExample() {
+        // Example from main() of TrialOfTheGrasses.java
+        TrialOfTheGrasses.TreeNode<Integer> nodes = new TrialOfTheGrasses.TreeNode<>(1,
+                new TrialOfTheGrasses.TreeNode<>(2,
+                        new TrialOfTheGrasses.TreeNode<>(3, new TrialOfTheGrasses.TreeNode<>(4),
```
Comment (Anatoly03) [inline w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java:15]:
```suggestion
                        new TrialOfTheGrasses.TreeNode<>(3,
                                new TrialOfTheGrasses.TreeNode<>(4),
```
formatting

File: w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                                new TrialOfTheGrasses.TreeNode<>(5))),
+                new TrialOfTheGrasses.TreeNode<>(5),
+                new TrialOfTheGrasses.TreeNode<>(6, new TrialOfTheGrasses.TreeNode<>(7),
```
Comment (Anatoly03) [inline w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java:18]:
```suggestion
                new TrialOfTheGrasses.TreeNode<>(6,
                        new TrialOfTheGrasses.TreeNode<>(7),
```

**Comment by Anatoly03** [comment]
Minor things can be changed, rest lgtm!

File: w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java (hunk 3 above)
Comment (nilsreichardt) [reply w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java:18]: Is this the formatting from your IDE?

**Review by felixhauptmann — APPROVED** [review]
Apart from Anatolys suggestions LGTM

File: w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java (hunk 3 above)
Comment (Anatoly03) [reply w10h01/test/pgdp/trials/TrailOfTheGrassesTest.java:18]: No, this is just for simplicity so every node is in it's line.

**Pull request merged**